src/cargo/ops/cargo_rustc/context.rs

```diff
@@ -191,9 +191,12 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
             Vacant(entry) => { entry.insert(req); }
         };
 
-        for &(pkg, dep, profile) in self.dep_targets(pkg, target, profile).iter() {
-            self.build_requirements(pkg, dep, profile, req);
-        }
+        req.each_kind(|kind| {
```

Comment:
I think that this function may benefit from actually just taking a `Kind` instead of `req: Platform`. That should avoid this extra loop and should make things a little more clear I think